mm2src/common/password_policy.rs

```diff
@@ -0,0 +1,116 @@
+use derive_more::Display;
+use regex::Regex;
+
+pub const PASSWORD_MAXIMUM_CONSECUTIVE_CHARACTERS: usize = 3;
+
+#[derive(Debug, Display, PartialEq)]
+pub enum PasswordPolicyError {
+    #[display(fmt = "Password can't contain the word password")]
+    ContainsTheWordPassword,
+    #[display(fmt = "Password length should be at least 8 characters long")]
+    PasswordLength,
+    #[display(fmt = "Password should contain at least 1 digit")]
+    PasswordMissDigit,
+    #[display(fmt = "Password should contain at least 1 lowercase character")]
+    PasswordMissLowercase,
+    #[display(fmt = "Password should contain at least 1 uppercase character")]
+    PasswordMissUppercase,
+    #[display(fmt = "Password should contain at least 1 special character")]
+    PasswordMissSpecialCharacter,
+    #[display(fmt = "Password can't contain the same character 3 times in a row")]
+    PasswordConsecutiveCharactersExceeded,
+}
+
+pub fn password_policy(password: &str) -> Result<(), PasswordPolicyError> {
+    lazy_static! {
+        static ref REGEX_NUMBER: Regex = Regex::new(".*[0-9].*").unwrap();
+        static ref REGEX_LOWERCASE: Regex = Regex::new(".*[a-z].*").unwrap();
+        static ref REGEX_UPPERCASE: Regex = Regex::new(".*[A-Z].*").unwrap();
+        static ref REGEX_SPECIFIC_CHARS: Regex = Regex::new(".*[^A-Za-z0-9].*").unwrap();
+    }
+    if password.to_lowercase().contains("password") {
```

Comment:
what if it contains "passWord" or "PASSword" or even "p@ssword" ? I think this is worth checking too!

Reply:
8 chars, should never pass anyway, regardless of password complexity.

Reply:
![image](https://user-images.githubusercontent.com/801971/227934030-fca258af-b68b-445a-b218-07dda851a5c8.png)